### PR TITLE
Fix "unintended_html_in_doc_comment" analysis errors

### PR DESCRIPTION
**Pull request by brianquinlan**
---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/blob/main/docs/External-Package-Maintenance.md#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: pkgs/http/lib/src/base_response.dart [hunk 1]
```diff
@@ -128,8 +128,9 @@ var _headerSplitter = RegExp(r'[ \t]*,[ \t]*');
 /// "Set-Cookie: lang=en; Path=/foo/,SID=x23"
 /// would both be result in `response.headers` => "lang=en; Path=/foo/,SID=x23"
 ///
-/// The idea behind this regex is that ",<valid token>=" is more likely to
-/// start a new <cookie-pair> then be part of <path-value> or <extension-av>.
+/// The idea behind this regex is that `,<valid token>=` is more likely to
+/// start a new `<cookie-pair>` then be part of `<path-value>` or
```
Comment (Anikate-De) [inline pkgs/http/lib/src/base_response.dart:132]:
```suggestion
/// start a new `<cookie-pair>` than be part of `<path-value>` or
```

File: pkgs/http_profile/lib/src/utils.dart [hunk 2]
```diff
@@ -10,13 +10,13 @@ const _tokenChars = r"!#$%&'*+\-.0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ^_`"
 /// Splits comma-separated header values.
 var _headerSplitter = RegExp(r'[ \t]*,[ \t]*');
 
-/// Splits comma-separated "Set-Cookie" header values.
+/// Splits comma-seperated "Set-Cookie" header values.
```
Comment (Anikate-De) [inline pkgs/http_profile/lib/src/utils.dart:13]: nit:
```suggestion
/// Splits comma-separated "Set-Cookie" header values.
```

File: pkgs/http_profile/lib/src/utils.dart [hunk 3]
```diff
@@ -25,8 +25,9 @@ var _headerSplitter = RegExp(r'[ \t]*,[ \t]*');
 /// "Set-Cookie: lang=en; Path=/foo/,SID=x23"
 /// would both be result in `response.headers` => "lang=en; Path=/foo/,SID=x23"
 ///
-/// The idea behind this regex is that ",<valid token>=" is more likely to
-/// start a new <cookie-pair> then be part of <path-value> or <extension-av>.
+/// The idea behind this regex is that `,<valid token>=` is more likely to
+/// start a new `<cookie-pair>` then be part of `<path-value>` or
```
Comment (Anikate-De) [inline pkgs/http_profile/lib/src/utils.dart:29]: nit:
```suggestion
/// start a new `<cookie-pair>` than be part of `<path-value>` or
```

**Review by Anikate-De — CHANGES_REQUESTED** [review]

**Review by Anikate-De — APPROVED** [review]

**Comment by natebosch** [comment]
@devoncarew This PR highlights a false positive for the lint. It doesn't understand square braces well enough and it looks like it flags `[List<int>]`. I think the change to ``` `List<int>` ``` is an improvement, but only slight compared to the obvious improvement from broken HTML. I didn't click on any of these cases when I skimmed the internal results, and the rollout is more churn than I expected.

I also forgot about the unpublished packages where they're unlikely to be noticed since we don't have any workflow where they are visible to users. I don't know whether any IDEs try to render the dartdoc to html for display, but that's the only use case I can imagine is affected.

**Review by natebosch — APPROVED** [review]
http and http_profile can both be bumped to the next version with `-wip`.

**Comment by devoncarew** [comment]
> @devoncarew This PR highlights a false positive for the lint. It doesn't understand square braces well enough and it looks like it flags `[List<int>]`. I think the change to `` `List<int>` `` is an improvement, but only slight compared to the obvious improvement from broken HTML. I didn't click on any of these cases when I skimmed the internal results, and the rollout is more churn than I expected.

Gotcha; can you file an issue in the sdk for this? We have had some other recent false positives (https://github.com/dart-lang/sdk/issues/56450) addressed - now fixed in head.

**Comment by natebosch** [comment]
https://github.com/dart-lang/sdk/issues/56465

**Comment by brianquinlan** [comment]
> http and http_profile can both be bumped to the next version with `-wip`.

Done

**Review by natebosch — APPROVED** [review]

**Pull request merged**